packages/ember-data/lib/system/store.js

```diff
@@ -1045,18 +1045,11 @@ Store = Service.extend({
   */
   all: function(modelName) {
     var typeClass = this.modelFor(modelName);
-    var typeMap = this.typeMapFor(typeClass);
-    var findAllCache = typeMap.findAllCache;
 
-    if (findAllCache) {
-      this.recordArrayManager.updateFilter(findAllCache, typeClass);
-      return findAllCache;
-    }
-
-    var array = this.recordArrayManager.createRecordArray(typeClass);
+    var liveRecordArray = this.recordArrayManager.liveRecordArrayFor(typeClass);
+    this.recordArrayManager.updateFilter(liveRecordArray, typeClass);
```

Comment:
I think it's because in the test creating a record eventually would trigger a `recordWasLoaded` which in turn updates the record arrays – but this event is triggered [in the `actions` queue](https://github.com/emberjs/data/blob/37401a8380cc9e6c26e8ad701c4b2f92d34edf3d/packages/ember-data/lib/system/model/internal-model.js#L200), which would be executed in the next step of the run loop... I guess :thought_balloon: